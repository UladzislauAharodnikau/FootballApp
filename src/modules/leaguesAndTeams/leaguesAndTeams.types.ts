import {LeaguesAndTeamsRoutes} from '@constants/routes.types';

export type LeaguesAndTeamsParamList = {
  [LeaguesAndTeamsRoutes.LeaguesAndTeamsHome]: undefined;
  [LeaguesAndTeamsRoutes.LeaguesList]: undefined;
  [LeaguesAndTeamsRoutes.LeagueDetails]: {id: number; leagueName: string};
  [LeaguesAndTeamsRoutes.TeamList]: {
    leagueId: number;
    seasonYear: number;
    leagueName: string;
  };
  [LeaguesAndTeamsRoutes.TeamDetails]: {teamId: number; leagueId: number};
};
