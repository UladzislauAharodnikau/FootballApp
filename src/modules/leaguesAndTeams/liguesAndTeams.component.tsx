import React from 'react';
import {
  createStackNavigator,
  StackNavigationOptions,
} from '@react-navigation/stack';
import {LeaguesAndTeamsRoutes} from '@constants/routes.types';
import {getStackScreenOptions} from 'shared/utils/getStackScreenOptions';
import LeaguesAndTeamsHome from 'modules/leaguesAndTeams/screens/leaguesAndTeamsHome/leaguesAndTeamsHome.component';
import {LeaguesAndTeamsParamList} from 'modules/leaguesAndTeams/leaguesAndTeams.types';
import LeaguesList from 'modules/leaguesAndTeams/screens/leaguesList/leaguesList.component';
import LeagueDetails from 'modules/leaguesAndTeams/screens/leagueDetails/leagueDetails.component';
import TeamList from 'modules/leaguesAndTeams/screens/teamList/teamList.component';
import TeamDetails from 'modules/leaguesAndTeams/screens/teamDetails/teamDetails.component';

const LeaguesAndTeamsStack = createStackNavigator<LeaguesAndTeamsParamList>();

const screenOptions = ({
  title,
  headerShown = false,
  headerBackTitleVisible = false,
}: StackNavigationOptions) => ({
  headerShown,
  title,
  headerBackTitleVisible,
});

const LeaguesAndTeamsNavigator = () => {
  return (
    <LeaguesAndTeamsStack.Navigator screenOptions={getStackScreenOptions()}>
      <LeaguesAndTeamsStack.Screen
        name={LeaguesAndTeamsRoutes.LeaguesAndTeamsHome}
        component={LeaguesAndTeamsHome}
      />
      <LeaguesAndTeamsStack.Screen
        name={LeaguesAndTeamsRoutes.LeaguesList}
        component={LeaguesList}
        options={screenOptions({title: 'Leagues', headerShown: true})}
      />
      <LeaguesAndTeamsStack.Screen
        name={LeaguesAndTeamsRoutes.LeagueDetails}
        component={LeagueDetails}
        options={screenOptions({title: '', headerShown: true})}
      />
      <LeaguesAndTeamsStack.Screen
        name={LeaguesAndTeamsRoutes.TeamList}
        component={TeamList}
        options={screenOptions({title: '', headerShown: true})}
      />
      <LeaguesAndTeamsStack.Screen
        name={LeaguesAndTeamsRoutes.TeamDetails}
        component={TeamDetails}
        options={screenOptions({title: '', headerShown: true})}
      />
    </LeaguesAndTeamsStack.Navigator>
  );
};

export default LeaguesAndTeamsNavigator;
