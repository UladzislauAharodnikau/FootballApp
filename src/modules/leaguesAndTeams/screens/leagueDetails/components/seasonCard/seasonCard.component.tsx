import React, {memo} from 'react';
import {Pressable, StyleSheet, Text} from 'react-native';
import {SeasonType} from 'shared/types/leagueItem.types';
import {testProps} from 'shared/utils/testProps';

interface SeasonCardProps {
  season: SeasonType;
  onSeasonCardPress(): void;
}

export const SeasonCard: React.FC<SeasonCardProps> = memo(
  ({season, onSeasonCardPress}) => {
    return (
      <Pressable
        {...testProps('season-card-press-id')}
        onPress={onSeasonCardPress}
        style={styles.container}>
        <Text>{season?.year}</Text>
      </Pressable>
    );
  },
);

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 10,
    paddingVertical: 5,
  },
});
