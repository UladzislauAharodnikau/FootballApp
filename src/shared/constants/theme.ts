export const Theme = {
  purple: '#673ab7',
  dark: '#222',
  gray: '#808080',
  lightGray: '#dcd9de',
  white: '#fff',
};
