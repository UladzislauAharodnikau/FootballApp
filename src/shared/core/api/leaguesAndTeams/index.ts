import instance from '../instance';

const LeaguesAndTeamsAPI = {
  getLeagues: () => instance.get('/leagues'),
  getLeagueTeams: (leagueId: number, seasonYear: number) =>
    instance.get(`/teams?league=${leagueId}&season=${seasonYear}`),
};

export default LeaguesAndTeamsAPI;
