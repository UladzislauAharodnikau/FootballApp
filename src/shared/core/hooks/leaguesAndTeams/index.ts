import {useQuery, useQueryClient} from 'react-query';
import LeaguesAndTeamsAPI from 'shared/core/api/leaguesAndTeams';
import {LeaguesAndTeamsQueryKeys} from 'shared/types/queryKeys.types';
import {AxiosResponse} from 'axios';
import {useEffect, useState} from 'react';
import {LeagueItem} from 'shared/types/leagueItem.types';
import {TeamItem} from 'shared/types/teamItem.types';
import {StatisticsType} from 'shared/types/teamStatistics.types';

type AxiosRequestType<T> = AxiosResponse<{response: T}>;
type QueryStateType<R> = {data: {response: R}};

type StatisticsParamsType = {
  seasonYear: number;
  teamId: number;
  leagueId: number;
};

const handleError = (err: unknown) => console.log(err?.message);

export const useGetLeagues = () => {
  const {data, error, isError, isLoading, isSuccess} = useQuery<
    AxiosRequestType<LeagueItem[]>
  >({
    queryKey: LeaguesAndTeamsQueryKeys.Leagues,
    queryFn: LeaguesAndTeamsAPI.getLeagues,
    onError: handleError,
  });

  return {
    data: data?.data.response ?? [],
    error,
    isError,
    isLoading,
    isSuccess,
  };
};

export const useGetLeagueById = (leagueId: number): LeagueItem | null => {
  const queryClient = useQueryClient();

  const [selectedLeague, setSelectedLeague] = useState<LeagueItem | null>(null);

  useEffect(() => {
    const state = queryClient.getQueryState<QueryStateType<LeagueItem[]>>(
      LeaguesAndTeamsQueryKeys.Leagues,
    )?.data?.data?.response;

    const leagueItem = state?.find(({league}) => league?.id === leagueId);

    setSelectedLeague(leagueItem ?? null);
  }, [leagueId, queryClient]);

  return selectedLeague;
};

export const useGetTeams = (leagueId: number, seasonYear: number) => {
  const {data, isError, error, isLoading} = useQuery<
    AxiosRequestType<TeamItem[]>
  >({
    queryKey: [LeaguesAndTeamsQueryKeys.Teams, leagueId],
    queryFn: () => LeaguesAndTeamsAPI.getLeagueTeams(leagueId, seasonYear),
    onError: handleError,
  });

  return {data: data?.data.response, error, isError, isLoading};
};

export const useGetTeamById = (
  teamId: number,
  leagueId: number,
): TeamItem | null => {
  const queryClient = useQueryClient();

  const [team, setTeam] = useState<TeamItem | null>(null);

  useEffect(() => {
    const state = queryClient.getQueryState<QueryStateType<TeamItem[]>>([
      LeaguesAndTeamsQueryKeys.Teams,
      leagueId,
    ])?.data?.data.response;

    const selectedTeam = state?.find(teamItem => teamItem.team.id === teamId);

    setTeam(selectedTeam ?? null);
  }, [leagueId, queryClient, teamId]);

  return team;
};

export const useGetStatistics = ({
  seasonYear,
  teamId,
  leagueId,
}: StatisticsParamsType) => {
  const {data, isError, error, isLoading} = useQuery<
    AxiosRequestType<StatisticsType>
  >({
    queryKey: LeaguesAndTeamsQueryKeys.Statistics,
    queryFn: () =>
      LeaguesAndTeamsAPI.getTeamStatistics({seasonYear, teamId, leagueId}),
    onError: handleError,
  });

  return {data: data?.data.response, isError, isLoading, error};
};
