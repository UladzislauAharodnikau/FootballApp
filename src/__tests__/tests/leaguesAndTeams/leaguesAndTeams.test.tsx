import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import {useGetLeagues, useGetTeams} from 'shared/core/hooks/leaguesAndTeams';
import LeaguesDetails from 'modules/leaguesAndTeams/screens/leaguesList/leaguesList.component';
import {Theme} from '@constants/theme';
import {mockedLeagues} from '__tests__/__mocks__/leagues.data';
import {LeagueCardList} from 'modules/leaguesAndTeams/screens/leaguesList/components/leagueCardList/leagueCardList.component';
import {LeagueCard} from 'modules/leaguesAndTeams/screens/leaguesList/components/leagueCard/leagueCard.component';
import {SeasonList} from 'modules/leaguesAndTeams/screens/leagueDetails/components/seasonList/seasonList.component';
import {seasonList} from '__tests__/__mocks__/season.data';
import {SeasonCard} from 'modules/leaguesAndTeams/screens/leagueDetails/components/seasonCard/seasonCard.component';
import TeamList from 'modules/leaguesAndTeams/screens/teamList/teamList.component';
import {teamsMockedData} from '__tests__/__mocks__/teams.data';
import {TeamCard} from 'modules/leaguesAndTeams/screens/teamList/components/teamCard/teamCard.component';

type GetLeaguesParamsType = {
  isLoading?: boolean;
  isError?: boolean;
  data?: any;
  error?: any;
};

describe('leagues and teams useGetLeagues', () => {
  beforeEach(() => {
    useGetLeagues.mockImplementation(() => ({
      isLoading: false,
      isError: false,
      data: [{}, {}],
      error: null,
    }));
  });

  test('test returned params', () => {
    const {data, isError, isLoading, error} = useGetLeagues();

    expect(isLoading).toBe(false);
    expect(error).toBe(null);
    expect(isError).toBe(false);

    expect(data).toBeTruthy();
    expect(data.length).toBe(2);

    expect(useGetLeagues).toHaveBeenCalled();
  });
});

describe('leagueList component test', () => {
  const getLeaguesCallback = ({
    isLoading = false,
    isError = false,
    data = [],
    error = null,
  }: GetLeaguesParamsType) =>
    useGetLeagues.mockImplementation(() => ({
      isLoading,
      isError,
      data,
      error,
    }));

  test('should be ActivityIndicator render when isLoading in true state, create snapshot', async () => {
    getLeaguesCallback({isLoading: true});

    const renderComponent = render(<LeaguesDetails />);

    const tree = renderComponent.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('snapshot when list is empty', async () => {
    getLeaguesCallback({});

    const renderComponent = render(<LeaguesDetails />);

    const tree = renderComponent.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('snapshot when list has items', async () => {
    getLeaguesCallback({data: mockedLeagues});

    const renderComponent = render(<LeaguesDetails />);

    const tree = renderComponent.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should be available ActivityIndicator when isLoading state true', () => {
    getLeaguesCallback({isLoading: true});

    const renderComponent = render(<LeaguesDetails />);

    const activityIndicatorComponent =
      renderComponent.getByTestId('activity-indicator');

    expect(activityIndicatorComponent).toBeTruthy();

    expect(activityIndicatorComponent.props.size).toBe(20);
    expect(activityIndicatorComponent.props.color).toBe(Theme.purple);
  });

  test('list of leagues snapshot', () => {});

  test('should render correct data into list of leagues', () => {
    const renderComponent = render(<LeagueCardList data={mockedLeagues} />);

    const leaguesList = renderComponent.getByTestId('leagues-list-id');

    expect(leaguesList).toBeTruthy();
    expect(leaguesList.props.data).toEqual(mockedLeagues);
  });
});

describe('League card component test', () => {
  let onPress: any, touchableItem: any;

  beforeEach(() => {
    onPress = jest.fn();

    const renderComponent = render(
      <LeagueCard league={mockedLeagues[0]} onLeagueCardPress={onPress} />,
    );

    touchableItem = renderComponent.getByTestId('card-league-press-id');
  });

  test('on press event callback should be pressed', () => {
    fireEvent.press(touchableItem);

    expect(onPress).toHaveBeenCalledTimes(1);

    fireEvent.press(touchableItem);

    expect(onPress).toHaveBeenCalledTimes(2);
  });

  test('should be called with valid ids which should be taken from item', () => {
    fireEvent.press(touchableItem);

    expect(onPress).toHaveBeenCalledWith(
      mockedLeagues[0].league.id,
      mockedLeagues[0].league.name,
    );
  });
});

// season list tests

describe('season list test', () => {
  test('season list snapshot with data', () => {
    const tree = render(
      <SeasonList data={seasonList} leagueName="League 1" leagueId={25} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('season list snapshot with empty data', () => {
    const tree = render(
      <SeasonList data={[]} leagueName="League 1" leagueId={25} />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('list item should have 4 items', () => {
    const tree = render(
      <SeasonList data={seasonList} leagueName="League 1" leagueId={25} />,
    );

    const component = tree.getByTestId('season-list-id');

    expect(component).toBeTruthy();

    expect(component.props.data.length).toBe(4);
  });

  test('season list card callback should call once', () => {
    const onSeasonCardPress = jest.fn();

    const cardComponent = render(
      <SeasonCard
        season={seasonList[0]}
        onSeasonCardPress={onSeasonCardPress}
      />,
    );

    const seasonCardTouchableItem = cardComponent.getByTestId(
      'season-card-press-id',
    );

    expect(seasonCardTouchableItem).toBeTruthy();

    fireEvent.press(seasonCardTouchableItem);

    expect(onSeasonCardPress).toHaveBeenCalledTimes(1);

    fireEvent.press(seasonCardTouchableItem);
    fireEvent.press(seasonCardTouchableItem);

    expect(onSeasonCardPress).toHaveBeenCalledTimes(3);

    //should be called without any params
    expect(onSeasonCardPress).toHaveBeenCalledWith();
  });
});

describe('team list', () => {
  const getTeamsCallback = ({
    isLoading = false,
    isError = false,
    data = [],
    error = null,
  }: GetLeaguesParamsType) =>
    useGetTeams.mockImplementation(() => ({
      isLoading,
      isError,
      data,
      error,
    }));

  test('should be available only activityIndicator component with isLoading true state', () => {
    getTeamsCallback({isLoading: true});

    const indicator = render(<TeamList />).getByTestId('activity-indicator');

    expect(indicator).toBeTruthy();

    expect(indicator.props.size).toBe(20);
    expect(indicator.props.color).toBe(Theme.purple);
  });

  test('team list snapshot when isLoading state in true', () => {
    getTeamsCallback({isLoading: true});

    const tree = render(<TeamList />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('team list snapshot when data is empty', () => {
    getTeamsCallback({});

    const tree = render(<TeamList />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('team list snapshot with data', () => {
    getTeamsCallback({data: teamsMockedData});

    const tree = render(<TeamList />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('team list should include 2 items', () => {
    getTeamsCallback({data: teamsMockedData});

    const component = render(<TeamList />).getByTestId('team-list-id');

    expect(component).toBeTruthy();

    expect(component.props.data.length).toBe(2);
  });

  test('team card should  be pressed and existed, and also called without params', () => {
    const onTeamCardPress = jest.fn();

    const teamCard = render(
      <TeamCard
        team={teamsMockedData[0].team}
        onTeamCardPress={onTeamCardPress}
      />,
    );

    const pressItem = teamCard.getByTestId('team-card-press-id');

    expect(pressItem).toBeTruthy();

    fireEvent.press(pressItem);

    expect(onTeamCardPress).toHaveBeenCalledTimes(1);
    expect(onTeamCardPress).toHaveBeenCalledWith();
  });
});
