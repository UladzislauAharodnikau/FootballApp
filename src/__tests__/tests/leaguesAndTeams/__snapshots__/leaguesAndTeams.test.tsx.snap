// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leagueList component test should be ActivityIndicator render when isLoading in true state, create snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
        "paddingHorizontal": 15,
      },
      {
        "paddingHorizontal": 0,
        "paddingTop": 10,
      },
    ]
  }
>
  <ActivityIndicator
    accessibilityLabel="activity-indicator"
    accessible={true}
    color="#673ab7"
    size={20}
    testID="activity-indicator"
  />
</View>
`;

exports[`leagueList component test snapshot when list has items 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
        "paddingHorizontal": 15,
      },
      {
        "paddingHorizontal": 0,
        "paddingTop": 10,
      },
    ]
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    accessibilityLabel="leagues-list-id"
    accessible={true}
    data={
      [
        {
          "country": {
            "code": "34",
            "flag": "",
            "name": "Test Country",
          },
          "league": {
            "id": 1,
            "logo": "",
            "name": "Test 1",
            "type": "League",
          },
          "seasons": [
            {
              "current": false,
              "end": Date { NaN },
              "start": Date { NaN },
              "year": 2020,
            },
          ],
        },
        {
          "country": {
            "code": "35",
            "flag": "",
            "name": "Test Country 2",
          },
          "league": {
            "id": 2,
            "logo": "",
            "name": "Test 2",
            "type": "League",
          },
          "seasons": [
            {
              "current": false,
              "end": Date { NaN },
              "start": Date { NaN },
              "year": 2022,
            },
          ],
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="leagues-list-id"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityLabel="card-league-press-id"
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
          testID="card-league-press-id"
        >
          <Image
            resizeMode="contain"
            source={
              {
                "uri": "",
              }
            }
            style={
              {
                "height": 65,
                "marginRight": 20,
                "width": 45,
              }
            }
          />
          <Text>
            Test 1
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#808080",
              "height": 1,
              "opacity": 0.5,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityLabel="card-league-press-id"
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
          testID="card-league-press-id"
        >
          <Image
            resizeMode="contain"
            source={
              {
                "uri": "",
              }
            }
            style={
              {
                "height": 65,
                "marginRight": 20,
                "width": 45,
              }
            }
          />
          <Text>
            Test 2
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`leagueList component test snapshot when list is empty 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
        "flex": 1,
        "paddingHorizontal": 15,
      },
      {
        "paddingHorizontal": 0,
        "paddingTop": 10,
      },
    ]
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    accessibilityLabel="leagues-list-id"
    accessible={true}
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="leagues-list-id"
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`season list test season list snapshot with data 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={[Function]}
  ListFooterComponent={[Function]}
  accessibilityLabel="season-list-id"
  accessible={true}
  data={
    [
      {
        "current": false,
        "end": Date { NaN },
        "start": Date { NaN },
        "year": 2022,
      },
      {
        "current": true,
        "end": Date { NaN },
        "start": Date { NaN },
        "year": 2023,
      },
      {
        "current": false,
        "end": Date { NaN },
        "start": Date { NaN },
        "year": 2024,
      },
      {
        "current": false,
        "end": Date { NaN },
        "start": Date { NaN },
        "year": 2025,
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  testID="season-list-id"
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityLabel="season-card-press-id"
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
        testID="season-card-press-id"
      >
        <Text>
          2022
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#808080",
            "height": 1,
            "opacity": 0.5,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityLabel="season-card-press-id"
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
        testID="season-card-press-id"
      >
        <Text>
          2023
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#808080",
            "height": 1,
            "opacity": 0.5,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityLabel="season-card-press-id"
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
        testID="season-card-press-id"
      >
        <Text>
          2024
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#808080",
            "height": 1,
            "opacity": 0.5,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityLabel="season-card-press-id"
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
        testID="season-card-press-id"
      >
        <Text>
          2025
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          {
            "backgroundColor": "#808080",
            "height": 1,
            "opacity": 0.5,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`season list test season list snapshot with empty data 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={[Function]}
  ListFooterComponent={[Function]}
  accessibilityLabel="season-list-id"
  accessible={true}
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  testID="season-list-id"
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View>
      <Text>
        List is empty
      </Text>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          {
            "backgroundColor": "#808080",
            "height": 1,
            "opacity": 0.5,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`team list team list snapshot when data is empty 1`] = `
<RNCSafeAreaView
  edges={
    [
      "right",
      "left",
      "top",
    ]
  }
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    accessibilityLabel="team-list-id"
    accessible={true}
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="team-list-id"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View>
        <Text>
          List of teams is empty
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`team list team list snapshot when isLoading state in true 1`] = `
<RNCSafeAreaView
  edges={
    [
      "right",
      "left",
      "top",
    ]
  }
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <ActivityIndicator
    accessibilityLabel="activity-indicator"
    accessible={true}
    color="#673ab7"
    size={20}
    testID="activity-indicator"
  />
</RNCSafeAreaView>
`;

exports[`team list team list snapshot with data 1`] = `
<RNCSafeAreaView
  edges={
    [
      "right",
      "left",
      "top",
    ]
  }
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    accessibilityLabel="team-list-id"
    accessible={true}
    data={
      [
        {
          "team": {
            "code": "22",
            "country": "England",
            "founded": 1900,
            "id": 1,
            "logo": "",
            "name": "Liverpool",
            "national": false,
          },
          "venue": {
            "address": "Test address 1",
            "capacity": 44,
            "city": "Test city",
            "id": 1,
            "image": "",
            "name": "Venue Test 1",
            "surface": "Test surface",
          },
        },
        {
          "team": {
            "code": "22",
            "country": "England",
            "founded": 1900,
            "id": 2,
            "logo": "",
            "name": "Arsenal",
            "national": false,
          },
          "venue": {
            "address": "Test address 2",
            "capacity": 43,
            "city": "Test city",
            "id": 2,
            "image": "",
            "name": "Venue Test 2",
            "surface": "Test surface",
          },
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="team-list-id"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityLabel="team-card-press-id"
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
          testID="team-card-press-id"
        >
          <Image
            resizeMode="contain"
            source={
              {
                "uri": "",
              }
            }
            style={
              {
                "height": 60,
                "marginRight": 20,
                "width": 40,
              }
            }
          />
          <Text>
            Liverpool
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#808080",
              "height": 1,
              "opacity": 0.5,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityLabel="team-card-press-id"
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
          testID="team-card-press-id"
        >
          <Image
            resizeMode="contain"
            source={
              {
                "uri": "",
              }
            }
            style={
              {
                "height": 60,
                "marginRight": 20,
                "width": 40,
              }
            }
          />
          <Text>
            Arsenal
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
